src/server/server.js

```diff
@@ -218,13 +219,35 @@ SimulationServer.prototype._streamAppHostHtml = function (request, response) {
                 return '<script src="' + scriptSource + '"></script>';
             }).join('');
 
-            // Note we replace "default-src 'self'" with "default-src 'self' ws:" (in Content Security Policy) so that
-            // websocket connections are allowed (this relies on a custom version of send that supports a 'transform' option).
+            // Note we add "connect-src 'self' ws:" and "img-src 'blob:' (in Content Security Policy) so that
+            // websocket connections are allowed and the camera plugin works (this relies on a custom version
+            // of send that supports a 'transform' option).
+            var metaTagRegex = /<\s*meta[^>]*>/;
+            var cspRegex = /http-equiv\s*=\s*(['"])Content-Security-Policy\1/;
+            var cspContent = /(content\s*=\s*")([^"]*)"/
             send(request, filePath, {
```

Comment:
nit: The call to `send` performs a critical task in order to make the simulation works: inject simulation files and makes an analyzisis of CSP in order to inject another configuration that allows connections to localhost for simulation. We could probably do a little refactor on `_streamAppHostHtml` and the promise chain at the prepare call. So it is easy to follow, smaller methods and more testable.
This is not needed to do it as part of this PR, it's just a comment :)